src/recon_buildblock/GeneralisedPrior.cxx

```diff
@@ -52,26 +52,37 @@ set_up(shared_ptr<const TargetT> const&)
 }
 
 template <typename TargetT>
-Succeeded 
+void
+GeneralisedPrior<TargetT>::
+compute_Hessian(TargetT& output,
+                const BasicCoordinate<3,int>& coords,
+                const TargetT& current_image_estimate) const
+{
+  if (this->is_convex())
+    error("GeneralisedPrior:\n  compute_Hessian implementation is not overloaded by your convex prior.");
+  else
+    error("GeneralisedPrior:\n  compute_Hessian is not implemented because the prior is not convex.");
```

Comment:
I'd rephrase this. a non-convex prior could be 2nd order differentiable of course.
```suggestion
    error("GeneralisedPrior:\n  compute_Hessian is not implemented for this (non-convex) prior.");
```